tests/xhUnitTest.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Unish;
+
+/**
+ * Unit tests for xh.drush.inc.
+ *
+ * @group base
+ */
+class xhUnitCase extends UnitUnishTestCase {
+
+  /**
+   * Test various combinations of XHProf flag options.
+   *
+   * @dataProvider xhOptionProvider
+   */
+  public function testFlags($name, $options, $expected) {
+    drush_preflight();
+    foreach ($options as $option_name => $option_value) {
+      drush_set_option($option_name, $option_value);
+    }
+    $this->assertEquals($expected, xh_flags(), $name);
+  }
+
+  /**
+   * Provides drush XHProf options and the results we expect from xh_flags().
+   */
+  public function xhOptionProvider() {
+
+    if (!defined('XHPROF_FLAGS_NO_BUILTINS')) {
+      define('XHPROF_FLAGS_NO_BUILTINS', 1);
+      define('XHPROF_FLAGS_CPU', 2);
+      define('XHPROF_FLAGS_MEMORY', 3);
```

Comment:
Added these defines because xhprof is not installed by default on Travis CI. Additionally xhprof extension is not available on PHP 7 yet.